src/BloomExe/web/ServerBase.cs

```diff
@@ -246,52 +211,125 @@ protected static string GetLocalPathWithoutQuery(IRequestInfo info)
 			return r;
 		}
 
-		protected void ReportMissingFile(IRequestInfo info)
+		public static string PathEndingInSlash
 		{
-			Logger.WriteEvent("**{0}: File Missing: {1}", GetType().Name, GetLocalPathWithoutQuery(info));
-			info.WriteError(404);
+			get { return "http://localhost:8089/bloom/"; }
 		}
 
-		protected virtual bool ProcessRequest(IRequestInfo info)
+		public static string GetContentType(string extension)
 		{
-			// process request for directory index
-			var requestedPath = info.LocalPathWithoutQuery.Substring(7);
-			if (info.RawUrl.EndsWith("/") && (Directory.Exists(requestedPath)))
+			switch (extension)
 			{
-				info.WriteError(403, "Directory listing denied");
-				return true;
+				case ".css": return "text/css";
+				case ".gif": return "image/gif";
+				case ".htm":
+				case ".html": return "text/html";
+				case ".jpg":
+				case ".jpeg": return "image/jpeg";
+				case ".js": return "application/x-javascript";
+				case ".png": return "image/png";
+				case ".pdf": return "application/pdf";
+				case ".txt": return "text/plain";
+				case ".svg": return "image/svg+xml";
+				default: return "application/octet-stream";
 			}
+		}
 
-			if (requestedPath.EndsWith("testconnection"))
+		/// <summary>
+		/// TODO: Note: doing this at runtime isn't as good as doing it in the installer, because we have no way of
+		/// removing these entries on uninstall (but the installer does).
+		/// </summary>
+		private static void AddUrlAccessControlEntry()
+		{
+			MessageBox.Show(
+				string.Format("We need to do one more thing before Bloom is ready. Bloom needs temporary administrator privileges to set up part of its communication with the embedded web browser.{0}{0}After you click 'OK', you may be asked to authorize this step.", Environment.NewLine),
+					@"Almost there!", MessageBoxButtons.OK);
+
+			var startInfo = new ProcessStartInfo { UseShellExecute = true, Verb = "runas" };
+
+			if (Environment.OSVersion.Version.Major == 5 /*win xp*/)
```

Comment:
We don't need special cases for WinXP any more. Bloom 3.1 and later simply won't run on XP.